src/lib/table/table.md

```diff
@@ -35,60 +48,89 @@ Here's a simple column definition with the name `'userName'`. The header cell co
-the names will define the order of the cells rendered. It is not required to provide a list of all 
-the defined column names, but only the ones that you want to have rendered.
+#### 3. Define the table's rows
 
-```html
-<mat-header-row *matHeaderRowDef="['userName', 'age']"></mat-header-row>
-<mat-row *matRowDef="let myRowData; columns: ['userName', 'age']"></mat-row>
-```
+Finally, once you have defined your columns, you need to tell the table which columns will be
+rendered in the header and data rows. 
 
-#### 3. Provide data
-
-The column and row definitions now capture _how_ data will render - all that's left is to provide 
-the data itself. For simple scenarios with client-side operations, `MatTableDataSource` offers a 
-quick and easy starting point. Simply create an instance of `MatTableDataSource` and set the items 
-to be displayed to the `data` property. For more advanced scenarios, applications will implement 
-one or more custom `DataSource` to capture more specific behaviors.
+To start, create a variable in your component that contains the list of the columns you want to
+render. 
 
 ```ts
-this.myDataSource = new MatTableDataSource();
-this.myDataSource.data = dataToRender;
+columnsToDisplay = ['userName', 'age'];
 ```
 
 ```html
-<mat-table [dataSource]=”myDataSource”>
-  ...
-</mat-table>
+<mat-header-row *matHeaderRowDef="columnsToDisplay"></mat-header-row>
+<mat-row *matRowDef="let myRowData; columns: columnsToDisplay"></mat-row>
 ```
 
+Note that this list of columns provided to the rows can be in any order, not necessary the order in
+which you wrote the column definitions. Also, you do not necessarily have to include every column
+that was defined in your template.
+
+This means that by changing your column list provided to the rows, you can easily re-order and
+include/exclude columns dynamically.  
+
+### Alternative ways to provide data
+
+#### Observable stream of data arrays
+
+An alternative approach to providing data to the table is by passing an Observable stream that emits
+the data array to be rendered each time it is changed. The table will listen to this stream and 
+automatically trigger an update to the rows each time a new data array is emitted.
+
+#### DataSource
+
+Also, another approach to providing data is by implementing a `DataSource`. This is simply
+an interface that has two functions: `connect` and `disconnect`. After passing a DataSource to the
+table, the table will call `connect()` to receive a stream that emits the data array that should be
+rendered. The table will call `disconnect` when the table is destroyed, which may be the right time
+to clean up any subscriptions that may have been registered during the `connect` process.
+
+The benefit to creating a `DataSource` class is that it can decouple rendering logic from your
+component and keep it encapsulated in a separate class (e.g. sorting, filtering, pagination, e.g.).
```

Comment:
It might be helpful to eventually have an example for each of these that is just like `table-basic`, but using the alternative approach.

Reply:
+1 I think this would be useful to demonstrate the differences, I'll make an issue following this PR getting merged